src/app/services/error.service.ts

```diff
@@ -0,0 +1,38 @@
+import { ErrorTexts } from "./../common/app.values";
+import { ApolloError } from "@apollo/client/core";
+import { Injectable } from "@angular/core";
+import { ModalService } from "../components/modal/modal.service";
+import { logError } from "../common/app.helpers";
+
+@Injectable({
+    providedIn: "root",
+})
+export class ErrorService {
+    private errorDescription = "";
+
+    constructor(private modalService: ModalService) {}
+
+    public processError(error: Error): void {
+        this.errorDescription = error.message;
+
+        if (error instanceof ApolloError) {
+            this.errorDescription = error.networkError
+                ? ErrorTexts.ERROR_NETWORK_DESCRIPTION
+                : ErrorTexts.ERROR_TECHNICAL_SUPPORT;
+        }
+        if (error.message === ErrorTexts.ERROR_SERVER_BAD_SQL_QUERY) {
```

Comment:
This piece of cod ewill quickly grow with new kinds of errors. We should have some kind of dictionary instead of `if`.

Reply:
Fixed